lib/spack/docs/containers.rst

```diff
@@ -591,6 +592,37 @@ the example below:
    COPY data /share/myapp/data
    {% endblock %}
 
+The template also defines a number of inner blocks for finer-grained control over the rendered Dockerfile:
+
+* `make_manifest`
+* `install_software`
+* `strip_binaries`
+* `make_activate`
+* `build_stage_extra_instructions`
+* `copy_into_final`
+* `make_entrypoint`
+* `final_stage_extra_instructions`
+* `set_entrypoint`
+
+An example using two of these inner blocks might look like this:
+
+.. code-block::
+   :emphasize-lines: 3-4,7
+
+   {% extends "container/Dockerfile" %}
+   {% block make_manifest %}
+   {{ super() }}
+   COPY custom_packages {{ paths.environment }}/custom_packages
+   {% endblock make_manifest %}
+   {% block final_stage_extra_instructions %}
+   COPY data /share/myapp/data
+   {% endblock %}
```

Comment:
Isn't this case equivalent to:
```jinja
{% extends "container/Dockerfile" %}
{% block build_stage %}
COPY custom_packages {{ paths.environment }}/custom_packages
{{ super() }}
{% endblock build_stage %}
{% block final_stage %}
{{ super() }}
COPY data /share/myapp/data
{% endblock %}
```
?[^1] Asking to check whether the example shows something that needs inner blocks.

[^1]: in the sense that the copy operation can be moved before creating the `spack.yaml`.

Reply:
No, your suggestion does not work.  Because `mkdir {{ paths.environment }}` will throw an error if `{{ paths.environment }}` already exists.

See the issue that motivates this PR at https://github.com/spack/spack/issues/41656

Reply:
Oh, that can be solved by using `mkdir -p` which is a much smaller fix

Reply:
Yes, that was my first suggestion but wasn't liked by @wdconinc 

It's a "much smaller fix" in the sense of fewer characters commited to git, but I think @wdconinc response was generally correct -- it potentially opens up a porential for unintended consequences.

I think *this* fix is "much smaller" in the sense that it will have zero side-effect unless someone decides to use the blocks.